src/components/Search/index.tsx

```diff
@@ -105,20 +110,43 @@ const handleSearchResult = async (
 const Search: FC<{
   content?: string
   hasButton?: boolean
+  truncateTypeHash?: boolean
   onEditEnd?: () => void
-}> = memo(({ content, hasButton, onEditEnd }) => {
+}> = memo(({ content, hasButton, onEditEnd, truncateTypeHash }) => {
+  const SEARCH_COUNT = 10
   const isMobile = useIsMobile()
   const { t } = useTranslation()
+  const [searchByType, setSearchByType] = useSearchType()
+  const [isSearchByName, setIsSearchByName] = useState(searchByType === 'name')
+  const [searchByNameResults, setSearchByNameResults] = useState<UDTQueryResult[] | null>(null)
```

Comment:
This looks like an unnecessary and non-standard implementation.

If the goal is to achieve `setSearchByNameResults(null)`, you can use the following approach instead:

```typescript
const queryClient = useQueryClient()
queryClient.resetQueries(['searchByName', searchValue])
```

Reply:
but `resetQueries` does not set `searchByNameResults` to null here, we need to clear the display content for user when click clear button

Reply:
I didn't make it clear. What I meant is that the existence of the `searchByNameResults` state is unnecessary.

You can directly use the state returned by `useQuery`.

Reply:
the result of `useQuery` may be empty, under that condition, the display message is like `oops, no records found`, and when the user clears input, there should be nothing displayed, thus, there are four states here: 

1. some search results
2. empty search results
3. don't search, clear inputs and displayed panel
4. loading...

only using `useQuery` is too simple to cover these states

Reply:
> * some search results
> * empty search results
> * don't search, clear inputs and displayed panel
> * loading...

* data?.map
* data == null
* `queryClient.resetQueries(['searchByName', searchValue]) -> data == null` or `{ status, fetchStatus, isFetched, ...someStatus } = useQuery(...)`
* { isLoading } = useQuery(...)

`useQuery` should be able to fully replace the capability of `useState` in the current use case because this state is currently only being used in the following places:

```tsx
setSearchByNameResults(null)
...
setSearchByNameResults(searchResult ? searchResult.slice(0, DISPLAY_COUNT) : [])
...
{searchByNameResults && <SearchByNameResults udtQueryResults={searchByNameResults} />}
```

Reply:
I got the point except for the 3rd option:

>  `queryClient.resetQueries(['searchByName', searchValue]) -> data == null or { status, fetchStatus, isFetched, ...someStatus } = useQuery(...)`

how do we distinguish between `empty seach result` and `do not search`

Reply:
> how do we distinguish between `empty seach result` and `do not search`

An empty search result does not necessarily mean that the API did not respond with anything.

It could mean that it returned an empty array, indicating either no search was performed or the search request is still in progress.

In such cases, the data would be `null`. However, if a search has been performed and the result is zero, the data would be an empty array `[]`.

Alternatively, you can also use various states provided by `useQuery`, such as `isFetched`, to determine if the request has been completed. The `resetQueries` function should correctly reset these states as well.

Reply:
> An empty search result does not necessarily mean that the API did not respond with anything.

I think this is exactly what I mean.

API call should always return a list, if nothing matches search condition, an empty array is returned, `[]` is different from `null`.

When user clicks `clear` button, he doesn't care about what the API call at all, just show nothing, so under this case, we can ignore `useQuery` and just set result to be `null`, hence we need to maintain a state like `searchByNameResults`, maybe I can rename it to `searchByNameState`

Reply:
Correct me if I'm wrong.
It seems that `null` stands for `search is not made`, while `[]` stands for `searched but empty returned`. It can be done by 
1. `search is not made`: initialize the query with the initial value `null`, and call `resetQuerys` to restore it when `reset` button is clicked;
2. `searched but empty returned` can be done by `returned []`;
3. `searched but thrown` can be done by `query.status === 'error'`(https://tanstack.com/query/v4/docs/react/guides/queries#query-basics);

Reply:
Correct, I suddenly realized why the communication was slow before. It might be because I forgot to provide detailed explanations:

`queryClient.resetQueries(['searchByName', searchValue])` will reset the entire state of `useQuery(['searchByName', searchValue])`, including but not limited to `data`, `error`, `isLoading`, `isFetched`, and so on, which are the return values of `searchQuery`.